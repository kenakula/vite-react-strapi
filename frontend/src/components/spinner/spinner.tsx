import { ReactElement } from 'react';

import styles from './spinner.module.css';

export const Spinner = (): ReactElement => (
  <div className={styles.ldsRoller} data-testid="spinner">
    <div></div>
    <div></div>
    <div></div>
    <div></div>
    <div></div>
    <div></div>
    <div></div>
    <div></div>
  </div>
);
