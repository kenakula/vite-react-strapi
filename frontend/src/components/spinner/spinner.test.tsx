import { screen } from '@testing-library/dom';
import { renderWithProviders } from '@tests/render-with-providers';
import { beforeEach, expect } from 'vitest';

import { Spinner } from './spinner';

const { getByTestId } = screen;

describe('SpinnerTest', () => {
  beforeEach(() => {
    renderWithProviders(<Spinner />);
  });

  test('should render spinner', () => {
    expect(getByTestId('spinner')).toBeVisible();
  });

  test('should match snapshot', () => {
    expect(getByTestId('spinner')).toMatchSnapshot();
  });
});
