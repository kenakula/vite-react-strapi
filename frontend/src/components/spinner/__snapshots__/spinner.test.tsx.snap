// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SpinnerTest > should match snapshot 1`] = `
<div
  class="uNyezq_ldsRoller"
  data-testid="spinner"
>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;
